app/src/main/java/com/epfl/drawyourpath/userProfile/cache/UserModelCached.kt

```diff
@@ -197,66 +189,28 @@ class UserModelCached(application: Application) : AndroidViewModel(application)
     }
 
     /**
-     * * // TODO: Code duplication for with updateActivityTimeGoal etc. refactor.
-     * Use this function to modify the daily distance goal of the user
+     * Use this function to modify the daily goals of the user
      * @param distanceGoal new daily distance goal
+     * @param activityTimeGoal new daily time goal
+     * @param pathsGoal new daily paths goal
      */
-    fun updateDistanceGoal(distanceGoal: Double): CompletableFuture<Unit> {
+    fun updateGoals(distanceGoal: Double? = null, activityTimeGoal: Double? = null, pathsGoal: Int? = null): CompletableFuture<Unit> {

```

Comment:
Should probably replace the argument list with the `UserGoals` data class to be more flexible.